molgenis-data-cache/src/main/java/org/molgenis/data/cache/L1/L1CacheRepositoryDecorator.java

```diff
@@ -0,0 +1,306 @@
+package org.molgenis.data.cache.L1;
+
+import autovalue.shaded.com.google.common.common.collect.Lists;
+import com.google.common.collect.Iterators;
+import org.molgenis.data.*;
+import org.molgenis.data.meta.model.EntityMetaData;
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.function.Consumer;
+import java.util.stream.Stream;
+
+import static com.google.common.collect.Iterators.partition;
+import static com.google.common.collect.Lists.newArrayList;
+import static java.util.Objects.requireNonNull;
+import static java.util.Spliterator.ORDERED;
+import static java.util.Spliterator.SORTED;
+import static java.util.Spliterators.spliteratorUnknownSize;
+import static java.util.stream.Collectors.toList;
+import static java.util.stream.Collectors.toMap;
+import static java.util.stream.StreamSupport.stream;
+import static org.molgenis.data.RepositoryCapability.CACHEABLE;
+import static org.molgenis.data.RepositoryCapability.WRITABLE;
+import static org.slf4j.LoggerFactory.getLogger;
+
+public class L1CacheRepositoryDecorator implements Repository<Entity>
```

Comment:
Add documentation describing what L1 cache implies and when data is cached or retrieved from cache.